### ISSUE TITLE
sails-mysql: boolean values sent as 0/1 when using .populate()

### ISSUE THREAD

**Issue by sgress454** [balderdashy/sails#3991]
_From @shobhitsinghal624 on August 8, 2014 14:23_

A boolean attribute of a model is sent as 0/1 instead of true/false when using .populate() although it works fine without the .populate. Here's how to replicate the issue.

Models -

``` .js
// api/models/Owner.js
module.exports = {
  attributes: {
    name:'STRING',
    age:'INTEGER',
    isAwesome:'boolean',
    pets:{
      collection: 'pet',
      via: 'owner'
    }
  }
}
```

``` .js
// api/models/Pet.js
module.exports = {
  attributes: {
    name:'STRING',
    color:'STRING',
    owner:{
      model:'owner'
    }
  }
}
```

Here are some `sails console` calls. The last call should point out the issue. It populates `isAwesome` field as `0/1` instead of `true/false` -

```
sails> Owner.create({name:'Mike',age:'21',isAwesome:true}).exec(console.log)
null { name: 'Mike',
  age: 21,
  isAwesome: true,
  createdAt: Fri Aug 08 2014 07:16:00 GMT-0700 (PDT),
  updatedAt: Fri Aug 08 2014 07:16:00 GMT-0700 (PDT),
  id: 1 }

sails> Pet.create({name:'Pinkie Pie',color:'pink',owner:1}).exec(console.log)
null { name: 'Pinkie Pie',
  color: 'pink',
  owner: 1,
  createdAt: Fri Aug 08 2014 07:18:22 GMT-0700 (PDT),
  updatedAt: Fri Aug 08 2014 07:18:22 GMT-0700 (PDT),
  id: 1 }

sails> Pet.create({name:'Applejack',color:'orange',owner:1}).exec(console.log)
null { name: 'Applejack',
  color: 'orange',
  owner: 1,
  createdAt: Fri Aug 08 2014 07:19:13 GMT-0700 (PDT),
  updatedAt: Fri Aug 08 2014 07:19:13 GMT-0700 (PDT),
  id: 2 }

sails> Owner.find().exec(console.log)
null [ { name: 'Mike',
    age: 21,
    isAwesome: true,
    id: 1,
    createdAt: Fri Aug 08 2014 07:16:00 GMT-0700 (PDT),
    updatedAt: Fri Aug 08 2014 07:16:00 GMT-0700 (PDT) } ]

sails> Owner.find().populate('pets').exec(console.log)
null [ { pets: 
     [ { name: 'Pinkie Pie',
         color: 'pink',
         owner: 1,
         id: 1,
         createdAt: Fri Aug 08 2014 07:18:22 GMT-0700 (PDT),
         updatedAt: Fri Aug 08 2014 07:18:22 GMT-0700 (PDT) },
       { name: 'Applejack',
         color: 'orange',
         owner: 1,
         id: 2,
         createdAt: Fri Aug 08 2014 07:19:13 GMT-0700 (PDT),
         updatedAt: Fri Aug 08 2014 07:19:13 GMT-0700 (PDT) } ],
    name: 'Mike',
    age: 21,
    isAwesome: 1,
    id: 1,
    createdAt: Fri Aug 08 2014 07:16:00 GMT-0700 (PDT),
    updatedAt: Fri Aug 08 2014 07:16:00 GMT-0700 (PDT) } ]
```

_Copied from original issue: balderdashy/sails-mysql#138_

**Comment by sgress454** [comment]
_From @particlebanana on December 8, 2014 20:20_

Hmm seems to not be casting it when associations are used. This might be more of a waterline issue but leaving here so it's more visible.

**Comment by sgress454** [comment]
_From @tarlepp on June 5, 2015 15:34_

So what is the status of this?

**Comment by sgress454** [comment]
_From @sailsbot on September 23, 2015 7:13_

Thanks for posting, @shobhitsinghal624.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:
- review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
- create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by sgress454** [comment]
_From @tarlepp on September 23, 2015 14:12_

And once again bot closes this issue without real response from core devs...

**Comment by sgress454** [comment]
_From @devinivy on September 23, 2015 14:51_

This is marked an an unresolved bug by @particlebanana, so it oughtta be open as far as I'm concerned.

**Comment by sgress454** [comment]
_From @sailsbot on October 24, 2015 6:27_

Thanks for posting, @shobhitsinghal624.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:
- review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
- create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by sgress454** [comment]
_From @tarlepp on October 24, 2015 11:22_

And again bot closes the issue that has not been solved...

**Comment by sgress454** [comment]
_From @tjwebb on October 25, 2015 3:40_

@sailsbot, take a hike

**Comment by sgress454** [comment]
_From @poacher2k on June 13, 2016 18:16_

@particlebanana Do you know if this has been taken care of in Waterline, or if it's an issue with sails-mysql?

**Comment by sgress454** [comment]
_From @tarlepp on June 15, 2016 20:34_

So what is the status of this issue today? It's over a year when I ask this first time, so getting some negative vibes about these issues and how these are handled...

**Comment by sgress454** [comment]
_From @animedbz16 on June 15, 2016 20:46_

Don't worry sailsbot will be here to sweep all the issues under the rug.

**Comment by sgress454** [comment]
_From @mikermcneil on June 20, 2016 18:43_

@poacher2k @shobhitsinghal624 @tarlepp @animedbz16 If you guys are still running into this issue, can you provide a bit more information?  Specifically, what does the underlying table look like?

It is definitely weird that we're seeing differences between when populate is or isn't being used.  The issue there is likely that a different method is run in sails-mysql when doing a populate vs. a regular find (in order to optimize the query).

The next step after seeing the underlying MySQL data type of the `isAwesome` column is to add a test to [Waterline adapter tests](https://github.com/balderdashy/waterline-adapter-tests) that replicates the bad behavior.

Thanks!

**Comment by sgress454** [comment]
_From @tarlepp on June 28, 2016 17:13_

I just tried to make this happen with latest version of sails, and I cannot reproduce this one. So my bet is that issue was fixed sometime within sails-mysql or waterline or sails itself.

**Comment by sgress454** [comment]
_From @tarlepp on June 28, 2016 17:20_

I also made simple application to test this behaviour which can be found there https://github.com/tarlepp/sails-mysql-populate-bug

Simple instructions:
- Add your database credentials to `config/local.js`
- Create your database
- Run command `sails console`
- Type `sails.models.owner.find().populate('pets').exec(console.log)` => [ENTER] => Results are just expected.
- Type `sails.models.pet.find().populate('owner').exec(console.log)` => [ENTER] => Results are just expected.
- Open your browser and try `http://localhost:1337/Test/test1` => Results are just expected.
- Open your browser and try `http://localhost:1337/Test/test2` => Results are just expected.

**Comment by sgress454** [comment]
@particlebanana I can reproduce this using @tarlepp's repo with the beta version of sails-mysql on Sails 1.0.  

I'll migrate this issue over to Sails momentarily, as we're consolidating everything in that repo.

**Comment by sgress454** [comment]
Confirmed fixed with latest beta version of sails-mysql on Sails 1.0.

**Comment by tarlepp** [comment]
finally!

**Comment by sgress454** [comment]
🎉This was incorporated into the official 1.0 release.
